test/bus.test.js

```diff
@@ -0,0 +1,613 @@
+		await db.users.destroy({
+			truncate: true,
+			restartIdentity: true,
+		});
+	});
+
+	it('should Retieve all permissions', async () => {
+		const response = await chai.request(app).get(`/api/v1/buses`);
+		expect(response).to.have.status(200);
+	});
+	it('should assign bus create permission to the operator role', async () => {
+		const response = await chai
+			.request(app)
+			.post(`/api/v1/roles/permission/assign`)
+			.send({
+				roleid: 2,
+				permissionid: 11,
+			});
+
+		expect(response).to.have.status(200);
+	});
+	it('should assign bus delete permission to the operator role', async () => {
+		const response = await chai
+			.request(app)
+			.post(`/api/v1/roles/permission/assign`)
+			.send({
+				roleid: 2,
+				permissionid: 12,
+			});
+
+		expect(response).to.have.status(200);
+	});
+	it('should assign bus update permission to the operator role', async () => {
+		const response = await chai
+			.request(app)
+			.post(`/api/v1/roles/permission/assign`)
+			.send({
+				roleid: 2,
+				permissionid: 13,
+			});
+
+		expect(response).to.have.status(200);
+	});
+	it('should assign create-route permission to the operator role', async () => {
+		const response = await chai
+			.request(app)
+			.post(`/api/v1/roles/permission/assign`)
+			.send({
+				roleid: 2,
+				permissionid: 6,
+			});
+
+		expect(response).to.have.status(200);
+	});
+	it('Should assign role ', (done) => {
+		chai
+			.request(app)
+			.put(`/api/v1/roles/assign/users`)
+			.send({
+				userId: 1,
+				roleId: 2,
+			})
+			.end((err, res) => {
+				chai.expect(res).to.have.status(200);
+				done();
+			});
+	});
+	it('Should access create route ', (done) => {
+		console.log('should assign role');
+		chai
+			.request(app)
+			.put(`/api/v1/roles/assign/users`)
+			.send({
+				userId: 1,
+				roleId: 2,
+			})
+			.end((err, res) => {
+				console.log('should should be permitted to access route');
+				chai
+					.request(app)
+					.post(`/api/v1/roles/createAccessTest`)
+					.set({
+						userId: 1,
+						action: 'createRoute',
+						'auth-token': 'Bearer ' + token,
+					})
+					.end((err, res) => {
+						chai.expect(res).to.have.status(201);
+						console.log('should remove the role');
+						chai
+							.request(app)
+							.delete(`/api/v1/roles/remove/users`)
+							.send({
+								roleId: 2,
+								userId: 1,
+							})
+							.end((err, res) => {
+								chai.expect(res).to.have.status(200);
+								done();
+							});
+					});
+			});
+	});
+	it('Should not access create route without login ', (done) => {
+		chai
+			.request(app)
+			.post(`/api/v1/roles/createAccessTest`)
+			.set({
+				userId: 1,
+				action: 'createRoute',
+				'auth-token': 'Bearer' + token,
+			})
+			.end((err, res) => {
+				chai.expect(res).to.have.status(401);
+				done();
+			});
+	});
+	it('Should get all buses ', (done) => {
+		chai
+			.request(app)
+			.get(`/api/v1/buses`)
+			.end((err, res) => {
+				chai.expect(res).to.have.status(200);
```

Comment:
Same here, how do we know that we'll get all the buses or the right ones? How do we know we're returning data... for all I know, the controller could just be sending the status and the test will still pass which is very very wrong.